tests/test_SimJtagMaster.py

```diff
@@ -395,16 +390,55 @@ def jtag_tests(self):
         fifo_tap2_content = self.chip["fifo2"].get_data()
 
         self.chip["JTAG"].scan_ir([BYPASS, DEBUG])
-        self.chip["JTAG"].scan_dr(["1", "0" * 24 + "10101101"], words=15)
+        self.chip["JTAG"].scan_dr([BitLogic("1"), BitLogic("0" * 24 + "10101101")] * 15, words=15)
 
         fifo_tap1_content = self.chip["fifo1"].get_data()
         fifo_tap2_content = self.chip["fifo2"].get_data()
-        self.assertListEqual(
-            [int("0" * 24 + "10101101", 2)] * 15, list(fifo_tap1_content)
-        )
-        self.assertNotEqual(
-            [int("0" * 24 + "10101101", 2)] * 15, list(fifo_tap2_content)
-        )
+
+        self.assertListEqual([int("0" * 24 + "10101101", 2)] * 15, list(fifo_tap1_content))
+        self.assertNotEqual([int("0" * 24 + "10101101", 2)] * 15, list(fifo_tap2_content))
+
+        # TEST OF SENDING MULTIPLE WORDS BY WRITING DIRECTLY IN JTAG MODULE MEMORY
+        self.chip["fifo1"].get_data()
+        self.chip["fifo2"].get_data()
+        data = np.byte([0x01, 0x02, 0x03, 0x04,
+                      0x11, 0x12, 0x13, 0x14,
+                      0x21, 0x22, 0x23, 0x24,
+                      0x31, 0x32, 0x33, 0x34,
+                      0x41, 0x42, 0x43, 0x44])
+        word_size_bit = 32
+        word_size_byte = int(word_size_bit / 8)
+        word_count = 5
+
+        self.chip["JTAG"].scan_ir([DEBUG, DEBUG])
+
+        # The ring register (DEBUG register) is 32 bits long, so the data have to be arranged like this : [WORD1(dev1) WORD1(dev2) WORD2(dev1) WORD2(dev2) ...]
+        ba = BitLogic()
+        all_data = bytearray()
+        for word in range(word_count):
+            ba.frombytes(data[word * word_size_byte:(word + 1) * word_size_byte].tobytes())
+            ba.fill()
+            ba.reverse()
```

Comment:
Why you need BitLogic here? Can you just `self.chip["JTAG"].set_data(data)`?

Why we have to reverse here?

Reply:
I use the Bitlogic only to reverse the data. We need to reverse because I write directly into the memory and the MSB and LSB is reversed in hardware right? Like in Verilog we have reg [32:0] and in python we have reg [0:32].
Do you know a better way to achieve this ?

Reply:
We should then in hardware go from the back for the word?

Reply:
You mean instead of reversing the word in python, we should read in reverse in the hardware ?

Reply:
Yes.

Here: https://github.com/SiLab-Bonn/basil/blob/6d72e7464a2cccb67eef5ca640748ce6f47fb2ad/basil/firmware/modules/jtag_master/jtag_master_core.v#L168

Somethng like:
`assign memout_addrb = (out_word_cnt * CONF_BIT_OUT) + CONF_BIT_OUT - 1 -out_bit_cnt;`

?